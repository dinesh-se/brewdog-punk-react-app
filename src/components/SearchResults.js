import React from 'react';

export default class SearchResults extends React.Component {
  render() {
    return (
      <div>
        <ul>
          {this.props.beerList.map((beer) => (
            <li key={beer.id}>{beer.name}</li>
          ))}
        </ul>
      </div>
    )
  }
}