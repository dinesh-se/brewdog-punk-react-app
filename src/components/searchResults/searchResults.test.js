import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from './searchResults';

test('should render list of beer correctly', () => {
  const beerList = [{
    id: 1,
    name: 'Beer 1',
    description: 'description 1',
    image_url: 'image1.png'
  }, {
    id: 2,
    name: 'Beer 2',
    description: 'description 2',
    image_url: 'image2.png'
  }];
  const wrapper = shallow(<SearchResults beerList={beerList} />);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.beer-search-results li')).toHaveLength(2);
  expect(wrapper.find('.no-results')).toHaveLength(0);
});

test('should display no results when no beers where found', () => {
  const beerList = [];
  const wrapper = shallow(<SearchResults beerList={beerList} />);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.no-results').text()).toBe('No search results found.');
  expect(wrapper.find('.beer-search-results')).toHaveLength(0);
});