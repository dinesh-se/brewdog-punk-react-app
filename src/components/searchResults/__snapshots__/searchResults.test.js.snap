// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display no results when no beers where found 1`] = `
<p
  className="no-results"
>
  No search results found.
</p>
`;

exports[`should render list of beer correctly 1`] = `
<div>
  <h3>
    Search Results
  </h3>
  <ul
    className="beer-search-results"
  >
    <li
      key="1"
    >
      <BeerItem
        beer={
          Object {
            "description": "description 1",
            "id": 1,
            "image_url": "image1.png",
            "name": "Beer 1",
          }
        }
      />
    </li>
    <li
      key="2"
    >
      <BeerItem
        beer={
          Object {
            "description": "description 2",
            "id": 2,
            "image_url": "image2.png",
            "name": "Beer 2",
          }
        }
      />
    </li>
  </ul>
</div>
`;
